plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,645 @@
+                .filter(size -> size != UNBOUNDED_LENGTH)
+                .map(VarcharType::createVarcharType)
+                .orElse(createUnboundedVarcharType());
+        return ColumnMapping.sliceMapping(varcharType, varcharReadFunction(varcharType), varcharWriteFunction(), FULL_PUSHDOWN);
+    }
+
+    private static ColumnMapping charColumnMapping(Optional<Integer> charLength)
+    {
+        if (charLength.isEmpty() || charLength.get() > CharType.MAX_LENGTH) {
+            return varcharColumnMapping(charLength);
+        }
+        CharType charType = createCharType(charLength.get());
+        return ColumnMapping.sliceMapping(
+                charType,
+                charReadFunction(charType),
+                charWriteFunction());
+    }
+
+    public static LongReadFunction dateReadFunction()
+    {
+        return (resultSet, columnIndex) -> {
+            long localMillis = resultSet.getDate(columnIndex).getTime();
+            return MILLISECONDS.toDays(localMillis);
+        };
+    }
+
+    public static LongWriteFunction dateWriteFunction()
+    {
+        return (statement, index, value) -> {
+            long millis = DAYS.toMillis(value);
+            statement.setDate(index, new Date(millis));
+        };
+    }
+
+    @Override
+    public void renameTable(ConnectorSession session, JdbcTableHandle handle, SchemaTableName newTableName)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support rename table");
+    }
+
+    @Override
+    protected String createTableSql(RemoteTableName remoteTableName, List<String> columns, ConnectorTableMetadata tableMetadata)
+    {
+        return createTableSql(quoted(remoteTableName), columns, tableMetadata.getProperties());
+    }
+
+    private String createTableSql(String remoteTableName, List<String> columns, Map<String, Object> tableProperties)
+    {
+        ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
+        ImmutableList.Builder<String> columnDefinitions = ImmutableList.builder();
+        columnDefinitions.addAll(columns);
+
+        List<String> primaryKeys = IgniteTableProperties.getPrimaryKey(tableProperties);
+        checkArgument(primaryKeys != null && !primaryKeys.isEmpty(), "No primary key defined for create table");
+        checkArgument(primaryKeys.size() < columns.size(), "Ignite table must have at least one non PRIMARY KEY column.");
+        columnDefinitions.add("PRIMARY KEY (" + join(", ", primaryKeys.stream().map(this::quoted).collect(joining(", "))) + ")");
+
+        for (Map.Entry<String, Object> propertyEntry : tableProperties.entrySet()) {
+            String propertyKey = propertyEntry.getKey();
+            switch (propertyKey) {
+                case PRIMARY_KEY_PROPERTY:
+                    break;
+                case IgniteTableProperties.AFFINITY_KEY_PROPERTY:
+                    String affinityKey = IgniteTableProperties.getAffinityKey(tableProperties);
```

Comment:
Does this value need to be quoted correctly to prevent injection?

Reply:
All the values will follow the `with` key word quoted by `"`. Ignite won't treat the properties as a part of sql.